iree/test/e2e/regression/generate_e2e_matmul_tests.py

```diff
@@ -263,14 +268,37 @@ class TestInputMatricesStaticShapes:
 # converts from the runtime shape dimensions in TestShape and given dynamicity to
 # the set of static shapes to be used in a test function's input tensors.
 def generate_static_shapes(shape: TestShape, dynamicity: Dynamicity):
-  return TestInputMatricesStaticShapes(
+  shapes = TestInputMatricesStaticShapes(
       lhs_rows=static_size(shape.m, dynamicity),
       lhs_cols=static_size(shape.k, dynamicity),
       rhs_rows=static_size(shape.k, dynamicity),
       rhs_cols=static_size(shape.n, dynamicity),
       acc_rows=static_size(shape.m, dynamicity),
       acc_cols=static_size(shape.n, dynamicity),
   )
+  # In the mixed-shapes case, we have just randomly picked each of the above 6
+  # values independently, making it likely that we got discrepancies where some
+  # of the M, K, N dimensions of the problem appear as a static dim in some
+  # matrix and as a dynamic dim in another matrix, e.g. for the M dimension we
+  # might have lhs_rows=dynamic, acc_rows=3. We should be testing both such
+  # 'wild' mixed-shapes cases, and more 'tame' mixed-shapes cases where each of
+  # the M, K, N dimensions is consistently either static or dynamic in both
+  # matrices (among lhs, rhs, acc) where it appears. If we don't do anything
+  # about it, as there is a 1/2 chance of discrepancy for each of the M, K, N
+  # dimensions, 7/8 of the mixed-shapes testcases will be 'wild'. Given our
+  # limited number of overall mixed-shapes testcases, this risks not testing
+  # the 'tame' case at all.
```

Comment:
I would note that shape inference should also be able to determine that these match and make them equivalent

Reply:
ah, ok. if you know how to make shape inference handle this, that sounds like an interesting way out here.

Reply:
I'd have to investigate where in the pipeline we do shape inference (if anywhere right now). Just in principle it definitely should be able to though :-P The dimensions have to match!